src/components/v2/devtronStackManager/DevtronStackManager.tsx

```diff
@@ -0,0 +1,450 @@
+                 */
+                if (allModulesErrorRes?.code >= 0) {
+                    setStackDetails({
+                        ...stackDetails,
+                        isLoading: false,
+                        errorStatusCode: allModulesErrorRes.code,
+                    })
+                    return
+                }
+
+                const logPodNameRes: LogPodNameResponse = responses[1].value
+                const releaseNotesRes: ReleaseNotesResponse = responses[2].value
+
+                const _stackDetails: StackDetailsType = {
+                    ...stackDetails,
+                    isLoading: false,
+                    releaseNotes: releaseNotesRes?.result
+                }
+
+                setLogPodName(logPodNameRes?.result?.podName)
+                pollForLatestDetails(allModulesRes?.result, _stackDetails)
+            })
+            .catch((err) => {
+                handleError(err)
+                setStackDetails({
+                    ...stackDetails,
+                    isLoading: false,
+                })
+            })
+    }
+
+    // Activate polling for latest server info, module details & logPodName only on stack manager page.
+    const pollForLatestDetails = (modulesList: ModuleDetails[], _stackDetails: StackDetailsType) => {
+        _getDetailsForAllModules(modulesList, _stackDetails)
+
+        // Fetching latest details/status every 30s
+        modulesPollingInterval = setInterval(() => {
+            getCurrentServerInfo()
+            _getDetailsForAllModules(modulesList, _stackDetails)
+            _getLogPodName()
+        }, 30000)
+    }
+
+    /**
+     * This is to handle the module selection
+     */
+    const handleModuleSelection = (moduleDetails: ModuleDetails, fromDiscoverModules?: boolean) => {
+        queryParams.set('id', moduleDetails.name)
+        setDetailsMode(fromDiscoverModules ? 'discover' : 'installed')
+        setSelectedModule(moduleDetails)
+
+        history.push(
+            `${
+                fromDiscoverModules
+                    ? URLS.STACK_MANAGER_DISCOVER_MODULES_DETAILS
+                    : URLS.STACK_MANAGER_INSTALLED_MODULES_DETAILS
+            }?${queryParams.toString()}`,
+        )
+    }
+
+    const handleBreadcrumbClick = () => {
+        setDetailsMode('')
+        setSelectedModule(undefined)
+    }
+
+    const handleTabChange = useCallback(
+        (tabIndex: number) => {
+            setSelectedTabIndex(tabIndex)
+        },
+        [selectedTabIndex],
+    )
+
+    const handleActionTrigger = useCallback(
+        (actionName: string, actionState: boolean) => {
+            setActionTriggered({
+                ...actionTriggered,
+                [actionName]: actionState,
+            })
+        },
+        [actionTriggered],
+    )
+
+    const checkIfCICDIsInstalled = () => {
+        return stackDetails.installedModulesList.findIndex((module) => module.name.toLowerCase() === 'cicd') !== -1
```

Comment:
Do we require to add 'includedinfullmode' check here as well?

Reply:
Nope. As it's already been checked while fetching module details so only installed items will be in this list.

Reply:
**Update** - removed the checked altogether.